[TECH] Ajoute la certification complementaire PRO_SANTE (PIX-12284)

aceol opened the pull request:
## :unicorn: Problème
On doit ajoute une nouvelle certification (PIX pro santé) mais l'interface ne le permet pas encore

## :robot: Proposition
Ajouter un script de migration

## :rainbow: Remarques
Il faut aussi raccorder les badges a cette certification complementaire. Ce sera gere dans une PR specifique

## :100: Pour tester
`select * from "complementary-certifications" where key = 'PRO_SANTE'`

aceol committed:
:card_file_box: api: Add new complementary certification PRO_SANTE

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr8754.review.pix.fr)
  * [App (.org)](https://app-pr8754.review.pix.org)
  * [Orga (.fr)](https://orga-pr8754.review.pix.fr)
  * [Orga (.org)](https://orga-pr8754.review.pix.org)
  * [Certif (.fr)](https://certif-pr8754.review.pix.fr)
  * [Certif (.org)](https://certif-pr8754.review.pix.org)
  * [1D](https://1d-pr8754.review.pix.fr)
  * [Admin](https://admin-pr8754.review.pix.fr)
  * [API](https://api-pr8754.review.pix.fr/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr8754.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr8754/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr8754/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr8754/environment)

Steph0 commented on api/db/migrations/20240425154143_add-pix-pro-sante.js:
```diff
@@ -0,0 +1,23 @@
+const key = 'PRO_SANTE';
+
+const up = async function (knex) {
+  const exists = await knex(TABLE_NAME).select(1).where({ key }).first();
```
Neatpick : Le first contient un select

```suggestion
  const exists = await knex(TABLE_NAME).first('id').where({ key });
```

ou bien

```suggestion
  const exists = await knex(TABLE_NAME).select(1).where({ key }).limit(1);
```

Steph0 replied:
Note : je serais + en faveur d'aller vers la gestion du 'onConflict'

aceol replied:
done

Steph0 commented on api/db/migrations/20240425154143_add-pix-pro-sante.js:
```diff
@@ -0,0 +1,23 @@
+
+const up = async function (knex) {
+  const exists = await knex(TABLE_NAME).select(1).where({ key }).first();
+  if (!exists) {
```
Ne faudrait-il pas quand même aligner les valeurs en cas d'existence ? Dans le sens où l'existence ne garantit pas que les valeurs soient bonnes (par exemple que minimumReproducibilityRate est bien à 45).

Plutôt passer par un `.onConflict(..).merge(updates)` qui assurera l'alignement des valeurs en cas d'existence ?

Documentation : https://knexjs.org/guide/query-builder.html#merge

Steph0 committed:
:recycle: api: we do not need to seed all types of complementary certifications

With the multiplication of complementary certifications, we only need to seed different types of complementary certifications, but not all of them

Steph0 reviewed (approved):
Neat!

aceol committed:
:bug: api: Fix tests with same key complementaries
